lib/pure/asyncnet.nim

```diff
@@ -130,6 +127,23 @@ type
     protocol: Protocol
   AsyncSocket* = ref AsyncSocketDesc
 
+proc hash*(x: AsyncSocket): Hash =
+  ## Generates a hash using only a subset of the fields.  The ones
+  ## used are considered to provide identity.
+  var h: Hash = 0
+  h = h !& hash(x.fd)
+  h = h !& hash(x.domain)
+  h = h !& hash(x.sockType)
+  h = h !& hash(x.protocol)
+  result = !$h
+
+proc `==`*(a, b: AsyncSocket): bool =
```

Comment:
But AsyncSocket is a ``ref`` type, it's not intuitive if equality for AsyncSocket is overwritten. We need a better solution.